### PR TITLE
Update PathFollow2D when curve is changed

### LINKED ISSUE #20386: Changing the position of a Path2D does not properly update the position of PathFollow2D nodes
**Godot version:**

3.0.5

**Issue description:**

Here's an image to show what I mean: https://i.imgur.com/OkT3h2D.png
The red dots are the ACTUAL positions of the path followers (they have random v and h offsets).
The blue line is the path with its **position currently set to (-16, 0).** 
The magenta line is the original path position.

The frogs are shown at the correct place but their actual positions are the original points of the path, instead of the points + position.

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #20386

I added a second commit (separately because I wasn't sure if it's necessary) that makes the update delayed, [as suggested by Calinou](https://github.com/godotengine/godot/issues/20386#issuecomment-538466693) 

Without delay
5g1UI0eBZu

With delay
1Z2ttsJmYJ

### CODE REVIEW

**Comment by Calinou** [comment]
If the slowdown with complex paths turns out to be an issue, maybe we should apply a dynamic delay that depends on the path complexity (e.g. 10 ms * number of points in the curve, with a delay of 500 ms at most).

File: scene/2d/path_2d.cpp
Comment (KiritoAM) [inline scene/2d/path_2d.cpp:238]: missing the 'f' suffix off the "0.5"; even tho the compiler will read it as a float anyway due to the param type

**Review by KiritoAM — CHANGES_REQUESTED** [review]

File: scene/2d/path_2d.cpp
Comment (KoBeWi) [reply scene/2d/path_2d.cpp:242]: It wasn't.

File: scene/2d/path_2d.cpp
Comment (KoBeWi) [reply scene/2d/path_2d.cpp:238]: f suffix isn't widely used in the codebase, so it's ok.

File: scene/2d/path_2d.h
Comment (KoBeWi) [reply scene/2d/path_2d.h:76]: Is padding even relevant in C++?

Comment (KiritoAM) [reply scene/2d/path_2d.h]: I feel it is when there could be many many instances of the class, as the wasted bytes from padding can add up. I can't really imagine 1000's of instances of this class being loaded at once, but I can imagine a game having a triple-digit quantity of the class. The wasted bytes from ~100 of this class alone isn't too bad at all, but it all adds up. I've experienced stuff like this meaning the difference between whether or not a mobile app is kicked out of memory by the OS.
Regardless, I'll leave it up to you regarding whether it's worth it or not - I've shared my opinions :)

**Review by KiritoAM — APPROVED** [review]

**Comment by Cynthia7979** [comment]
@KoBeWi Are you by any chance still working on this? I encountered the `PathFollow2D` updating issue (as in #20386) recently and it was a relief to see that someone had attempted to patch it. It'd be much appreciated if you're willing to resolve the conflicts once more and merge the branch.

**Comment by KoBeWi** [comment]
Rebased. It would be nice if someone reviewed the code.
Also we should decide whether go with delay or not.

**Comment by Cynthia7979** [comment]
> Rebased. It would be nice if someone reviewed the code. Also we should decide whether go with delay or not.

Personally I think it'd be the best if the delay is optional, probably configurable through a project setting or node attribute. However I'm not proficient with C++ so I'm not sure if that would be easy to implement.

**Comment by KoBeWi** [comment]
It would be easy to implement, but such option wouldn't be really useful I think. With Timer there is better performance, without Timer there's better responsiveness. I think it could be balanced if the Timer didn't restart with each move; you'd see updated positions every 0.5 seconds for example.

**Comment by KoBeWi** [comment]
Ok I updated the PR - the Timer is only started when it's not running. So the follow updates in real-time, but with 0.2s delay.
godot windows editor dev x86_64_Gpbnza3tP0

File: scene/2d/path_2d.cpp
Comment (akien-mga) [inline scene/2d/path_2d.cpp:256]: Keep an empty line between `_notification` cases, as those are typically pretty big, like standalone methods/callbacks.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**
